include/seqan3/argument_parser/validators.hpp

```diff
@@ -925,11 +925,12 @@ class regex_validator
      */
     template <std::ranges::forward_range range_type>
     //!\cond
-        requires std::convertible_to<std::ranges::range_value_t<range_type>, option_value_type const &>
+        requires std::convertible_to<std::ranges::range_value_t<range_type>, option_value_type const &> &&
+                 (!std::same_as<range_type, std::filesystem::path>)
     //!\endcond
     void operator()(range_type const & v) const
     {
-         std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+        std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
```

Comment:
You require `std::convertible_to<std::ranges::range_value_t<range_type>, option_value_type const &>` above, but you don't use that property anywhere in here.

https://en.cppreference.com/w/cpp/concepts/convertible_to says:

```c++
template <class From, class To>

concept convertible_to =
  std::is_convertible_v<From, To> &&
  requires(std::add_rvalue_reference_t<From> (&f)()) {
    static_cast<To>(f());
  };
```

so shouldn't you call the function like this?

```suggestion
        std::for_each(v.begin(), v.end(), [&] (auto cmp)
        {
            (*this)(static_cast<option_value_type const &>(cmp));
        });
```

?

Reply:
Yes that's better :)

Reply:
Well, despite making this more readable, convetible_to requires implicit conversion. So before it was fine as well.

Reply:
Not completely sure what you mean: here was the problem that this called recursively the same `operator()`, because `void operator()(range_type const & v) const` was more specific than the version of `operator()` that required the implicit cast.